pkg/controllers/v1alpha1/dataset/dataset_controller.go

```diff
@@ -203,11 +203,40 @@ func (r *DatasetReconciler) reconcileDatasetDeletion(ctx reconcileRequestContext
 
 	// 2. if there are datasets mounted this dataset, then requeue
 	if ctx.Dataset.Status.DatasetRef != nil && len(ctx.Dataset.Status.DatasetRef) > 0 {
```

Comment:
How about using `if len(ctx.Dataset.Status.DatasetRef) > 0` instead of  `ctx.Dataset.Status.DatasetRef != nil && len(ctx.Dataset.Status.DatasetRef) > 0`?

Reply:
ok.